### DISCUSSION TITLE
When I executed the test I got some errors： _ebpf_link_client_attach_provider returned error

### DISCUSSION THREAD

**Discussion by tower111** [microsoft/ebpf-for-windows#3286] [Q&A]
An error is reported during socket_tests.exe program execution. The error message captured is as follows. When running the test in https://github.com/microsoft/ebpf-for-windows-demo, it will also fail. The return result of the bpf_object__load function is -22. There is no problem with netsh ebpf add prog xxx.o. The problem occurs in Where, thank you very much
![image](https://github.com/microsoft/ebpf-for-windows/assets/31761342/dafefd22-c35e-49e9-9933-df5a46489e55)
![image](https://github.com/microsoft/ebpf-for-windows/assets/31761342/582869b8-0f59-473c-980f-91c4560b4f85)

**Comment by dahavey** [comment]
@saxena-anurag to have a look at this.

**Comment by tower111** [comment]
I changed the conn_track.sys loaded in ebpf-for-windows-demo to conn_track.o and it ran normally. I also disabled the windows mandatory signature. What could be wrong with this?
https://answers.microsoft.com/zh-hans/windows/forum/all/win10%E7%B3%BB%E7%BB%9F%E6%B0%B8%E4%B9%85%E7%A6% 81/8ae6a66a-b7bc-4835-8d18-483b27888488

The same question was asked here
https://github.com/microsoft/ebpf-for-windows-demo/issues/117
